### DISCUSSION TITLE
why is the image is colorful?

### DISCUSSION THREAD

**Discussion by bubble114** [mrdbourke/pytorch-deep-learning#703] [Q&A]
hello,
I am reading the course 1.2 Visualizing our data: https://www.learnpytorch.io/03_pytorch_computer_vision/#12-visualizing-our-data
since the image shape is Image shape: torch.Size([1, 28, 28]), it's should be gray. Why is the image showed by plt colorful?

**Comment by Shyamal-Dharia** [comment]
Hi @bubble114 , I'm not the expert here. but here what I have understood.

The default cmap value for plt.imshow() is "viridis". I believe that is the reason why you can see a colorful image at first, but when you use cmap = "gray" then the function uses grayscale mapping for the image.

here is the link for checking the default value for imshow() function: https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.imshow.html

**Reply by shisirkha** [reply]
Correct
